data_structures/src/Stack/Stack.cs

```diff
@@ -6,44 +6,61 @@ namespace ADT
 {
     public class Stack<T> : IStack<T>
     {
+        private T[] _items { get; set; }
+        private int _topOfStack { get; set; }
+
         public Stack(int size)
         {
-
+            _items = new T[size]; 
         }
 
         public IEnumerator<T> GetEnumerator()
         {
-            throw new NotImplementedException();
+             return (IEnumerator<T>)_items.GetEnumerator(); 
         }
 
         public bool IsEmpty()
         {
-            throw new NotImplementedException();
+            return Size() == 0;
         }
 
         public T Peek()
         {
-            throw new NotImplementedException();
+            if(!IsEmpty())            
+                return _items[_topOfStack -1];            
+            else            
+                throw new Exception("Stack is empty - Cannot peek.");            
```

Comment:
Just wanted to point out that it's better style to check for error conditions and return as soon as possible. So swapping the conditional to:

```cs
public T Peek()
{
	if(IsEmpty())
	    throw new Exception("Stack is empty - Cannot peek.");

	return _items[_topOfStack -1];
}
```
Basically the else should only be used for things that matter.